### ISSUE TITLE
[MemoryProfiler] Don't declare non built-in type variables in the header file

### ISSUE THREAD

**Issue by Jokeren** [GVProf/GVProf#32]
The problem appears for every new analysis header file

**Comment by Lin-Mao** [comment]
Do you mean I can't declare the user-defined data type (like [struct ChunkFragmentation](https://github.com/GVProf/redshow/blob/f8f3b59d0be12988a3f2ea09414ab133b299dba1/include/analysis/memory_profile.h#L158)) in the header file?

**Comment by Jokeren** [comment]
No, I meant the initialization shouldn't happen in the header file.

https://github.com/GVProf/redshow/blob/f8f3b59d0be12988a3f2ea09414ab133b299dba1/include/analysis/memory_profile.h#L130

**Comment by Lin-Mao** [comment]
OK. What is the drawback of that? I saw you also put the [Map<u64, std::shared_ptr<Memory>> _memories;](https://github.com/GVProf/redshow/blob/f8f3b59d0be12988a3f2ea09414ab133b299dba1/include/analysis/memory_profile.h#L114) in the header file.

**Comment by Lin-Mao** [comment]
Do you think there are too many global variables?

**Comment by Jokeren** [comment]
> OK. What is the drawback of that? I saw you also put the [Map<u64, std::shared_ptr> _memories;](https://github.com/GVProf/redshow/blob/f8f3b59d0be12988a3f2ea09414ab133b299dba1/include/analysis/memory_profile.h#L114) in the header file.

Can you point me any locations? I didn't write this file.

https://stackoverflow.com/questions/8108634/global-variables-in-header-file

> Do you think there are too many global variables?

No

**Comment by Jokeren** [comment]
You don't have to change the files right now. I just opened an issue to track this problem.

**Comment by Lin-Mao** [comment]
OK, got it.
